proxygen_cli/cli/command_credentials.py

```diff
@@ -20,37 +21,72 @@ def list():
     """
     List all credentials values.
     """
+    if not _yaml_credentials_file_source(None):
+        click.echo("Credentials file does not exist. Please run 'proxygen credentials set'")
+        exit()
+
     creds = get_credentials()
     output.print_spec(json.loads(creds.json(exclude_none=True)))
 
+
 @credentials.command()
 @click.argument("key", type=CHOICE_OF_CREDENTIAL_KEYS)
 def get(key):
     """
     Read a value from your credentials.
     """
+    if not _yaml_credentials_file_source(None):
+        click.echo("Credentials file does not exist. Please run 'proxygen credentials set'")
+        exit()
+
     creds = get_credentials()
     click.echo(getattr(creds, key))
 
 
 @credentials.command()
-@click.argument("key", type=CHOICE_OF_CREDENTIAL_KEYS)
-@click.argument("value")
-def set(key, value):
+@click.argument("custom_pairs", nargs=-1, metavar="KEY VALUE", required=False)
+@click.option("--force", is_flag=True, help="Force re-entry of standard credentials")
+def set(custom_pairs, force):
     """
     Write a value to your credentials.
     """
+    if not _yaml_credentials_file_source(None):
+        create_yaml_credentials_file()
+
     current_credentials = _yaml_credentials_file_source(None)
-    current_credentials[key] = value
+
+    # Check if base credentials are set
+    base_credentials_set = all(
+        current_credentials.get(field) is not None
+        for field in ["client_id", "client_secret", "username", "password"]
+    )
+
+    if not base_credentials_set or force:
+        client_id = click.prompt("Enter client_id")
+        client_secret = click.prompt("Enter client_secret")
+        username = click.prompt("Enter username", default="", show_default=False)
+        password = click.prompt("Enter password", default="", show_default=False)
+
+        current_credentials["client_id"] = client_id
+        current_credentials["client_secret"] = client_secret
+        current_credentials["username"] = username
+        current_credentials["password"] = password
+
+    # Prompt for individual custom key-value pairs
+    for i in range(0, len(custom_pairs), 2):
+        key, value = custom_pairs[i:i + 2]
```

Comment:
The reason for using [i:i+2] instead of [i:i+1] in the loop is because each iteration of the loop processes two elements at a time. The key and value.

Reply:
ah, ignore me - i wasn't really reading this right / was forgetting how slice indexing works